WIP: updates for chicken-5

corngood opened the pull request:
I don't know if this repo is still maintained, but I've been working on getting chicken-5 into nixpkgs, and as part of that I've updated egg2nix.

Here's what I've got so far.

corngood committed:
updates for chicken-5

the-kenny commented:
I’m not actively using chicken anymore, but I’m totally fine with updating this :) we can also talk about moving this to a shared organisation or transferring ownership completely. 

> Am 14.11.2018 um 02:50 schrieb David McFarland <notifications@github.com>:
> 
> I don't know if this repo is still maintained, but I've been working on getting chicken-5 into nixpkgs, and as part of that I've updated egg2nix.
> 
> Here's what I've got so far.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/the-kenny/egg2nix/pull/3
> 
> Commit Summary
> 
> updates for chicken-5
> File Changes
> 
> R egg2nix.egg (3)
> M egg2nix.scm (115)
> D egg2nix.setup (6)
> M eggs.nix (287)
> Patch Links:
> 
> https://github.com/the-kenny/egg2nix/pull/3.patch
> https://github.com/the-kenny/egg2nix/pull/3.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or mute the thread.

neeasade commented:
I think this fails on packages with a dash in the name -- https://ptpb.pw/XQIa

evhan committed:
Allow program to read dependencies from egg files

This lets `egg2nix' run with a local egg file or directory, in which
case it will read dependencies from the local egg specification. This
brings it more in line with `cabal2nix' and similar tools, in that
`egg2nix .` will generate a Nix expression from a local source tree.

Adds a test script with some basic checks around input file handling.
The `locate-egg-file' and `read-dependencies' procedures are also
exported for use in the tests, and the `main' entrypoint procedure is
only called when egg2nix.scm is compiled so that it can also be loaded
into the tests or a REPL.

evhan committed:
Add Travis CI configuration

evhan committed:
Run install step with "-sudo" in CI

evhan committed:
Switch Travis build to 5.0.1 snapshot

evhan committed:
Cache chicken-install dependencies in CI

corngood committed:
Merge pull request #1 from evhan/local-egg-file-input

Allow program to read dependencies from egg files

DerGuteMoritz committed:
Don't barf on C5 egg version specs

We don't want to interpret these as a pinned versions as they are
specified to be interpreted equal to or greater than.

DerGuteMoritz committed:
Don't pollute egg set with auxiliary functions

This could lead to collisions with eggs of the same name. Also, it
allows one to obtain all egg derivations via builtins.attrValues which
is handy for buildInputs in a shell.nix.

corngood committed:
Merge pull request #2 from DerGuteMoritz/chicken-5

Some improvements for chicken-5 branch
